### ISSUE TITLE
Problem during install java image into docker

### ISSUE THREAD

**Issue by ameydesai-tudip** [dockerfile/java#27]
I am having below issue during installation of java image in docker ? I want solution to resolve the same?

Sending build context to Docker daemon 99.33 kB
Step 1 : FROM dockerfile/java:oracle-java8
Pulling repository docker.io/dockerfile/java
Error: image dockerfile/java not found

**Comment by raskolnikov7** [comment]
I m facing same issue.

**Comment by raskolnikov7** [comment]
91m/bin/sh: 1: [0m[91madd-apt-repository: not found[0m[91m
[0mRemoving intermediate container b41d7e7b1c47
The command '/bin/sh -c echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | debconf-set-selections &&   add-apt-repository -y ppa:webupd8team/java &&   apt-get update &&   apt-get install -y oracle-java8-installer &&   rm -rf /var/lib/apt/lists/\* &&   rm -rf /var/cache/oracle-jdk8-installer' returned a non-zero code: 127

**Comment by demaniak** [comment]
Yes, it seems the dockerfile/java image is not available on [docker hub](https://hub.docker.com/r/dockerfile/java/) - it is 404'ing

**Comment by dhesson** [comment]
Now what?!

**Comment by dhesson** [comment]
I uploaded this image as a base for my java builds.  Seems to work.

https://hub.docker.com/r/cogniteev/oracle-java/~/dockerfile/

**Comment by demaniak** [comment]
@dhesson long story short, Oracle does not want the images there, so chances are your image will be taken down in the near future. Seems they do not want Oracle java to be distributed through any channels other than their own. Also I believe there is some feelings about that license acceptance that has to happen before java is installed.

The official java image repo (https://hub.docker.com/_/java/) also some explanation on this.

A work-a-round is to start up your own private registry, and host your base image in that for your builds. I have no idea where that leaves you with regards to Oracle (legally speaking).

**Comment by loretoparisi** [comment]
@demaniak this has been now deprecated, so which is the solution?

**Comment by demaniak** [comment]
@loretoparisi you mean [this](https://hub.docker.com/_/java/)?

I would suggest it is possible to role your own image, maybe using [this](https://github.com/dockerfile/java/blob/master/oracle-java8/Dockerfile) as a guide.

I would also suggest you then host that image in your [own private repository.](https://books.sonatype.com/nexus-book/3.0/reference/docker.html) 

Hope that helps, good luck!

**Comment by loretoparisi** [comment]
@demaniak thanks the issue was the old docker files of java, it's as simple as `docker pull java`
